### PR TITLE
Adding non-implemented functions in Complex

### PR DESCRIPTION
**Pull request by mafiya69**
some non-implemented overloads for `rdivcomp` added.

### CODE REVIEW

File: src/complex.h [hunk 1]
```diff
@@ -194,6 +194,34 @@ class Complex : public Number {
...
+            throw std::runtime_error("Divide by zero.");
+        } else {
+            return from_mpq((this->real_ * other.i) / conjugate, (this->imaginary_ * (-other.i)) / conjugate);
+        }
+    }
+
+    /*! Divide other by the Complex
+     * \param other of type Complex
+     * */
+     inline RCP<const Number> rdivcomp(const Complex &other) const {
```
Comment (isuruf) [inline src/complex.h]: We don't really need this method because of `divcomp(const Complex &other) const {` method of `Complex`

Comment (mafiya69) [inline src/complex.h]: @isuruf I will remove that, is everything else ok?

File: src/complex.h [hunk 2]
```diff
@@ -229,6 +242,8 @@ class Complex : public Number {
             return rsubcomp(static_cast<const Rational&>(other));
         } else if (is_a<Integer>(other)) {
             return rsubcomp(static_cast<const Integer&>(other));
+        } else if (is_a<Complex>(other)){
+            return rsubcomp(static_cast<const Complex&>(other));
```
Comment (isuruf) [inline src/complex.h]: These two lines are not needed as well

File: src/complex.h [hunk 3]
```diff
@@ -261,6 +276,10 @@ class Complex : public Number {
     virtual RCP<const Number> rdiv(const Number &other) const {
         if (is_a<Integer>(other)) {
             return rdivcomp(static_cast<const Integer&>(other));
+        } else if (is_a<Rational>(other)) {
+            return rdivcomp(static_cast<const Rational&>(other));
+        } else if (is_a<Complex>(other)) {
+            return (static_cast<const Complex&>(other).divcomp(*this));
```
Comment (isuruf) [inline src/complex.h]: Since we are not calling `rdiv` directly, lines 281-282 are not needed as well.

**Comment by mafiya69** [comment]
@isuruf

**Comment by certik** [comment]
This looks good, thanks. Can you add a test as well? That fails in master
and that your patch fixes.

**Comment by mafiya69** [comment]
@certik what does shippable basically do? I am asking because `ctest` is passing on my system.

**Comment by certik** [comment]
I think you can see for yourself in Shippable --- you just need to login using your github account (for example). I looked in there, looks like all tests pass except the last 3, which fail due to some Shippable bug -- it doesn't say why. So I wouldn't worry about it.

**Comment by mafiya69** [comment]
@certik will surely try to add tests for `Complex` as there are none

**Comment by mafiya69** [comment]
@certik I have added some tests, Have a look and let me know if I am doing something wrong.

**Comment by isuruf** [comment]
@mafiya69, 
Tests are here, https://github.com/sympy/symengine/blob/master/src/tests/basic/test_basic.cpp#L485

**Comment by mafiya69** [comment]
@isuruf as there are `divnum`, `subnum` etc global functions for `div`, `sub` etc. ([Here](https://github.com/sympy/symengine/blob/master/src/number.h#L57)), shouldn't there be `rdivnum`, `rsubnum` etc for `rdiv`, `rsub` etc. ?

**Comment by isuruf** [comment]
You could always use `divnum(y, x)` instead of `rdivnum(x, y)`. That's why they are not implemented.

**Comment by isuruf** [comment]
If you could add some tests that use, `Complex::rdivcomp(const Rational &other)` then we can merge this.

**Comment by mafiya69** [comment]
@isuruf

**Comment by isuruf** [comment]
+1

**Comment by Sumith1896** [comment]
@mafiya69 rebase needed, look into the updated test files in master to modify your tests.
